docs/proposals/kubectl-login.md

```diff
@@ -0,0 +1,233 @@
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<!-- TAG RELEASE_LINK, added by the munger automatically -->
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.3/docs/proposals/kubelet-eviction.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Kubectl Login Subcommand
+
+**Authors**: Eric Chiang (@ericchiang)
+
+## Goals
+
+`kubectl login` attempts to guide the user's experience when configuring
+credentials for kubectl. It should assist users who are first experiencing
+Kubernetes, users who are partially setup, and help users who are already
```

Comment:
perhaps split this into use-cases?

 1. Empty configuration, trying to connect to a server.
 2. Server specified, trying to authenticate.
 3. Trying to switch servers.  I've connected to serverA and I want to switch to serverB.
 5. Trying to reauthenticate as the same user because my credentials have expired.
 4. Authenticate as a different user to the same server.  Not common, but it happens.

Long term goal: enable the pluggable `AuthProvider` infrastructure to be discoverable from a master to avoid the end-user having to know how his sysadmin configured kube.